moveit_core/robot_model/src/planar_joint_model.cpp

```diff
@@ -208,31 +216,51 @@ void PlanarJointModel::interpolate(const double* from, const double* to, const d
     double dx, dy, initial_turn, drive_angle, final_turn;
     computeTurnDriveTurnGeometry(from, to, min_translational_distance_, dx, dy, initial_turn, drive_angle, final_turn);
 
+    // approximate cost (distance traveled) doing initial turn
     double initial_d = fabs(initial_turn) * angular_distance_weight_;
+    // approximate cost (distance traveled) doing straight drive
     double drive_d = hypot(dx, dy);
+    // approximate cost (distance traveled) doing final turn
     double final_d = fabs(final_turn) * angular_distance_weight_;
 
+    // total cost of executing manuever
     double total_d = initial_d + drive_d + final_d;
 
+    // If the difference between `from` and `to` is so low that it is within floating point arithmetic error
+    // or if `from == to`, the following operations will result in nan
+    // Just set the return state to target state and don't interpolate.
+    if (total_d == 0.0)
```

Comment:
The comment correctly states that you want to test for floating point accuracy like so:
```suggestion
    if (total_d < std::numeric_limits<float>::epsilon())
```

Reply:
Similar note as above. With the above change, the divisions in the following lines should technically work even if the angle change is very small. 

However, if we are ok with saying we won't support values this small, we can definitely use `std::numeric_limits<float>::epsilon()` as a threshold.